src/java.base/share/classes/java/io/Console.java

```diff
@@ -105,7 +104,8 @@ public class Console implements Flushable
     * @return  The printwriter associated with this console
     */
     public PrintWriter writer() {
-        return pw;
+        throw new UnsupportedOperationException(
+                "Console class itself does not provide implementation");
```

Comment:
Hello Naoto, should we perhaps then mark this method (and thus the class too) as `abstract` and leave the sub-classes to provide this method's implementation?

Reply:
Thanks for the review, Jai. You're right that making it `abstract` would eliminate implementations in `Console` class, but I was hesitant to do so, as it is a spec change that would suggest readers to think that `Console` is generally pluggable (which is not).